src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,691 @@
+*
+* This test case provides a scope to  get the State of FP Power LED.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel idicator state should be set to On through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 8 and provide indicator index.
+*
+*/
+void test_9_dsFPD_hal_GetLEDState(void)
+{
+    gTestID = 9;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    dsFPDLedState_t state = 0;
+
+    UT_LOG_INFO("Calling dsFPSetLEDState(OUT:state)");
+    status = dsFPGetLEDState(&state);
+    UT_LOG_INFO("Result dsGetFPState(OUT:FP LED State:[%X]) dsError_t:[%s]" \
+                                                ,state,UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+    /* Check that the Indicator is valid */
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to set LED indicator with suported colors.
+*
+* This test case provides a scope to  set the LED inicaotrs to different Supported Colors.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel indicator state should be set to ON through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 9 and provide indicator value and Color value.
+*
+*/
+void test_10_dsFPD_hal_SetFPColor(void)
+{
+    gTestID = 10;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+    int ret = 0;
+    dsError_t status = dsERR_NONE;
+    uint32_t uColor;
+    bool isValidColor = false;
+
+    ret = selectIndicator();
+    if(!ret)
+        goto exit;
+    UT_LOG_MENU_INFO(" \t  Supported Front Panel color for the indicator %s are:",UT_Control_GetMapString(dsFrontPanelIndicatorTable, eIndicator));
+    UT_LOG_MENU_INFO("------------------------------------------");
+
+    for (int j = 0; j < dsFPD_COLOR_MAX; j++)
+    {
+        UT_LOG_MENU_INFO("\t0X%06X.  %-20s\n", dsFrontPanelColorTable[j].value, dsFrontPanelColorTable[j].string);
+    }
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Select Color: ");
+    scanf("%u", &uColor);
+    readAndDiscardRestOfLine(stdin);
+    isValidColor = false;
+    for (int j = 0; j < dsFPD_COLOR_MAX; j++)
+    {
+                if( dsFrontPanelColorTable[j].value == uColor)
+                    isValidColor = true;
+    }
+    if(!isValidColor)
+    {
+        UT_LOG_INFO("Unsupported Color Input 0x%08X",uColor);
+        goto exit;
+    }
+
+    /* Check that the Indicator is valid */
+    UT_LOG_INFO("Calling SetFPColor(IN:Indicator:[0x%d], IN:Color:[%X])" \
```

Comment:
Function name is wrong in log string. Please check.

Reply:
updated